lib/client/doc.js

```diff
@@ -104,11 +139,24 @@ emitter.mixin(Doc);
 Doc.prototype.destroy = function(callback) {
   var doc = this;
   doc.whenNothingPending(function() {
-    doc.connection._destroyDoc(doc);
     if (doc.wantSubscribe) {
-      return doc.unsubscribe(callback);
+      doc.unsubscribe(function(err) {
```

Comment:
This would be a good place to use the ShareDB constructor presence flag to conditionally use the more complex `unsubscribe` handler. Here's the original commend from @ericyhwang & @nateps :

> Add a ShareDB constructor option/flag for enabling presence. It looks like currently, some things like op caching could affect performance, even if you never call doc.submitPresence. It'd be good to not run that type of code if you're not using presence, like if none of your doc types don't support it.